scripts/west_commands/runners/nrfutil.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for flashing with nrfutil.'''
+
+import json
+import os
+import sys
+import subprocess
+import tempfile
+
+from runners.nrf_common import NrfBinaryRunner
+
+class NrfUtilBinaryRunner(NrfBinaryRunner):
+    '''Runner front-end for nrfutil.'''
+
+    def __init__(self, cfg, family, softreset, dev_id, erase=False,
+                 tool_opt=[], force=False, recover=False):
+
+        super().__init__(cfg, family, softreset, dev_id, erase,
+                         tool_opt, force, recover)
+        self._ops = []
+        self._op_id = 1
+
+    @classmethod
+    def name(cls):
+        return 'nrfutil'
+
+    @classmethod
+    def tool_opt_help(cls) -> str:
+        return 'Additional options for nrfutil, e.g. "--log-level"'
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        return NrfUtilBinaryRunner(cfg, args.nrf_family, args.softreset,
+                                   args.dev_id, erase=args.erase,
+                                   tool_opt=args.tool_opt, force=args.force,
+                                   recover=args.recover)
+
+    def _exec(self, args):
+        try:
+            out = self.check_output(['nrfutil', '--json', 'device'] + args)
+        except subprocess.CalledProcessError as e:
```

Comment:
A little cleaner to use `as cpe` instead of `as e` followed by `cpe = e`

Reply:
I think that won't work:
https://docs.python.org/3/reference/compound_stmts.html#except-clause
> When an exception has been assigned using as target, it is cleared at the end of the except clause. This is as if

Reply:
Ah, that's why you left that link in the comment. OK